### PR TITLE
Implementing new ElectronicParkBrakeStatus

### LINKED ISSUE #632: [SDL 0102] New vehicle data - ElectronicParkBrakeStatus
Proposal: [New vehicle data - ElectronicParkBrakeStatus](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0102-New-vehicle-data-ElectronicParkBrakeStatus.md)

> This proposal is about adding a new vehicle data to get the status of the Park Brake from the Electric Park Brake (EPB) system.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/303

Steering Committee Decision:

> The Steering Committee fully agreed to accept this proposal and the enums defined in the [comment](https://github.com/smartdevicelink/sdl_evolution/issues/303#issuecomment-333927396) left by JackLivio on this review issue.

### PR DESCRIPTION
**Pull request by askirk**
Fixes #632 

This PR is **[ready]** for review.

### Risk
This PR makes **[minor]** API changes.

### Testing Plan
Unit tests were updated to cover new code changes.

### Summary
* adding new enum
* adding additional VehicleDataType
* adding field in Subscribe and Unsubscribe Vehicle Data Request & Response
* adding field in GetVehicleData Request & Response
* adding field in OnVehicleData notification
* adding corresponding sections to unit tests to cover new code

##### Enhancements
*  New vehicle data available `electronicParkBrakeStatus`

### CLA
- [X] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/enums/ElectronicParkBrakeStatusTests.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+			ElectronicParkBrakeStatus temp = ElectronicParkBrakeStatus.valueForString(example);
+			assertNull("Result of valueForString should be null.", temp);
+		}
+		catch (NullPointerException exception) {
+			fail("Null string throws NullPointerException.");
+		}
+	}
+
+	/**
+	 * Verifies the possible enum values of WiperStatus.
```
Comment (brettywhite) [inline sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/enums/ElectronicParkBrakeStatusTests.java]: Not WiperStatus

File: sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/requests/GetVehicleDataTests.java [hunk 2]
```diff
@@ -162,6 +165,7 @@ public void testRpcValues () {
         assertNull(Test.NULL, msg.getEmergencyEvent());
         assertNull(Test.NULL, msg.getClusterModeStatus());
         assertNull(Test.NULL, msg.getMyKey());
+	    assertNull(Test.NULL, msg.getElectronicParkBrakeStatus());
```
Comment (brettywhite) [inline sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/requests/GetVehicleDataTests.java]: Not sure how picky we want to be with tabs/spaces but there are a lot of lines in this PR that dont line up

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/GetVehicleData.java [hunk 3]
```diff
@@ -474,4 +475,12 @@ public Boolean getFuelRange() {
      * @return a Boolean value.
      */
     public Boolean getTurnSignal() { return getBoolean(KEY_TURN_SIGNAL); }
+
+    public void setElectronicParkBrakeStatus(Boolean electronicParkBrakeStatus){
```
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/GetVehicleData.java]: Please add javadoc

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/GetVehicleData.java [hunk 4]
```diff
@@ -474,4 +475,12 @@ public Boolean getFuelRange() {
      * @return a Boolean value.
      */
     public Boolean getTurnSignal() { return getBoolean(KEY_TURN_SIGNAL); }
+
+    public void setElectronicParkBrakeStatus(Boolean electronicParkBrakeStatus){
+        setParameters(KEY_ELECTRONIC_PARK_BRAKE_STATUS, electronicParkBrakeStatus);
+    }
+
+    public Boolean getElectronicParkBrakeStatus(){
```
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/GetVehicleData.java]: Please add javadoc

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/GetVehicleDataResponse.java [hunk 5]
```diff
@@ -295,4 +297,11 @@ public void setTurnSignal(TurnSignal turnSignal) {
     public TurnSignal getTurnSignal() {
         return (TurnSignal) getObject(TurnSignal.class, KEY_TURN_SIGNAL);
     }
+
+    public void setElectronicParkBrakeStatus(ElectronicParkBrakeStatus electronicParkBrakeStatus){
```
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/GetVehicleDataResponse.java]: Please add javadoc

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/GetVehicleDataResponse.java [hunk 6]
```diff
@@ -295,4 +297,11 @@ public void setTurnSignal(TurnSignal turnSignal) {
     public TurnSignal getTurnSignal() {
         return (TurnSignal) getObject(TurnSignal.class, KEY_TURN_SIGNAL);
     }
+
+    public void setElectronicParkBrakeStatus(ElectronicParkBrakeStatus electronicParkBrakeStatus){
+        setParameters(KEY_ELECTRONIC_PARK_BRAKE_STATUS, electronicParkBrakeStatus);
+    }
+    public ElectronicParkBrakeStatus getElectronicParkBrakeStatus(){
```
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/GetVehicleDataResponse.java]: Please add javadoc

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/OnVehicleData.java [hunk 7]
```diff
@@ -510,4 +512,11 @@ public void setTurnSignal(TurnSignal turnSignal) {
     public TurnSignal getTurnSignal() {
         return (TurnSignal) getObject(TurnSignal.class, KEY_TURN_SIGNAL);
     }
+
+    public void setElectronicParkBrakeStatus(ElectronicParkBrakeStatus electronicParkBrakeStatus){
```
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/OnVehicleData.java]: Please add javadoc

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/OnVehicleData.java [hunk 8]
```diff
@@ -510,4 +512,11 @@ public void setTurnSignal(TurnSignal turnSignal) {
     public TurnSignal getTurnSignal() {
         return (TurnSignal) getObject(TurnSignal.class, KEY_TURN_SIGNAL);
     }
+
+    public void setElectronicParkBrakeStatus(ElectronicParkBrakeStatus electronicParkBrakeStatus){
+        setParameters(KEY_ELECTRONIC_PARK_BRAKE_STATUS, electronicParkBrakeStatus);
+    }
+    public ElectronicParkBrakeStatus getElectronicParkBrakeStatus(){
```
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/OnVehicleData.java]: Please add javadoc

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/SubscribeVehicleData.java [hunk 9]
```diff
@@ -790,4 +791,11 @@ public Boolean getFuelRange() {
 	 * @return a Boolean value.
 	 */
 	public Boolean getTurnSignal() { return getBoolean(KEY_TURN_SIGNAL); }
+
+	public void setElectronicParkBrakeStatus(boolean electronicParkBrakeStatus){
```
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/SubscribeVehicleData.java]: Please add javadoc

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/SubscribeVehicleData.java [hunk 10]
```diff
@@ -790,4 +791,11 @@ public Boolean getFuelRange() {
 	 * @return a Boolean value.
 	 */
 	public Boolean getTurnSignal() { return getBoolean(KEY_TURN_SIGNAL); }
+
+	public void setElectronicParkBrakeStatus(boolean electronicParkBrakeStatus){
+		setParameters(KEY_ELECTRONIC_PARK_BRAKE_STATUS, electronicParkBrakeStatus);
+	}
+	public Boolean getElectronicParkBrakeStatus(){
```
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/SubscribeVehicleData.java]: Please add javadoc

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/SubscribeVehicleDataResponse.java [hunk 11]
```diff
@@ -449,4 +450,11 @@ public void setTurnSignal(VehicleDataResult turnSignal) {
     public VehicleDataResult getTurnSignal() {
         return (VehicleDataResult) getObject(VehicleDataResult.class, KEY_TURN_SIGNAL);
     }
+
+    public void setElectronicParkBrakeStatus(VehicleDataResult electronicParkBrakeStatus){
```
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/SubscribeVehicleDataResponse.java]: Please add javadoc

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/SubscribeVehicleDataResponse.java [hunk 12]
```diff
@@ -449,4 +450,11 @@ public void setTurnSignal(VehicleDataResult turnSignal) {
     public VehicleDataResult getTurnSignal() {
         return (VehicleDataResult) getObject(VehicleDataResult.class, KEY_TURN_SIGNAL);
     }
+
+    public void setElectronicParkBrakeStatus(VehicleDataResult electronicParkBrakeStatus){
+        setParameters(KEY_ELECTRONIC_PARK_BRAKE_STATUS, electronicParkBrakeStatus);
+    }
+    public VehicleDataResult getElectronicParkBrakeStatus(){
```
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/SubscribeVehicleDataResponse.java]: Please add javadoc

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/UnsubscribeVehicleData.java [hunk 13]
```diff
@@ -784,4 +785,11 @@ public Boolean getFuelRange() {
 	 * @return a Boolean value.
 	 */
 	public Boolean getTurnSignal() { return getBoolean(KEY_TURN_SIGNAL); }
+
+	public void setElectronicParkBrakeStatus(Boolean electronicParkBrakeStatus) {
```
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/UnsubscribeVehicleData.java]: Please add javadoc

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/UnsubscribeVehicleData.java [hunk 14]
```diff
@@ -784,4 +785,11 @@ public Boolean getFuelRange() {
 	 * @return a Boolean value.
 	 */
 	public Boolean getTurnSignal() { return getBoolean(KEY_TURN_SIGNAL); }
+
+	public void setElectronicParkBrakeStatus(Boolean electronicParkBrakeStatus) {
+		setParameters(KEY_ELECTRONIC_PARK_BRAKE_STATUS, electronicParkBrakeStatus);
+	}
+	public Boolean getElectronicParkBrakeStatus() {
```
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/UnsubscribeVehicleData.java]: Please add javadoc

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/UnsubscribeVehicleDataResponse.java [hunk 15]
```diff
@@ -454,4 +455,11 @@ public void setTurnSignal(VehicleDataResult turnSignal) {
     public VehicleDataResult getTurnSignal() {
         return (VehicleDataResult) getObject(VehicleDataResult.class, KEY_TURN_SIGNAL);
     }
+
+    public void setElectronicParkBrakeStatus(VehicleDataResult electronicParkBrakeStatus){
```
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/UnsubscribeVehicleDataResponse.java]: Please add javadoc

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/UnsubscribeVehicleDataResponse.java [hunk 16]
```diff
@@ -454,4 +455,11 @@ public void setTurnSignal(VehicleDataResult turnSignal) {
     public VehicleDataResult getTurnSignal() {
         return (VehicleDataResult) getObject(VehicleDataResult.class, KEY_TURN_SIGNAL);
     }
+
+    public void setElectronicParkBrakeStatus(VehicleDataResult electronicParkBrakeStatus){
+        setParameters(KEY_ELECTRONIC_PARK_BRAKE_STATUS, electronicParkBrakeStatus);
+    }
+    public VehicleDataResult getElectronicParkBrakeStatus(){
```
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/UnsubscribeVehicleDataResponse.java]: Please add javadoc

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/enums/ElectronicParkBrakeStatus.java [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+	 */
+	OPEN,
+	/**
+	 * When driver pulls the Electronic Park Brake switch while driving "at speed".
+	 */
+	DRIVE_ACTIVE,
+	/**
+	 * When system has a fault or is under maintenance.
+	 */
+	FAULT;
```
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/enums/ElectronicParkBrakeStatus.java]: Please add a comma, an extra clear line, and move the semicolon underneath

**Pull request merged**